specs/Standards/MultiToken/Metadata.md

```diff
@@ -0,0 +1,141 @@
+
+## Interface
+
+Metadata applies at both the class level (`MTBaseTokenMetadata`) and the specific instance level (`MTTokenMetadata`). The relevant metadata for each:
+
+```ts
+
+type MTContractMetadata = {
+  spec: string, // required, essentially a version like "mt-1.0.0"
+  name: string, // required Zoink's Digitial Sword Collection
+}
+
+type MTBaseTokenMetadata = {
+  name: string, // required, ex. "Silver Swords" or "Metaverse 3"
+  id: string, // required a unique identifier for the metadata
+  symbol: string|null, // required, ex. "MOCHI"
+  icon: string|null, // Data URL
+  decimals: string|null // number of decimals for the token useful for FT related tokens
+  base_uri: string|null, // Centralized gateway known to have reliable access to decentralized storage assets referenced by `reference` or `media` URLs
+  reference: string|null, // URL to a JSON file with more info
+  copies: number|null, // number of copies of this set of metadata in existence when token was minted.
+  reference_hash: string|null, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+type MTTokenMetadata = {
+  title: string|null, // ex. "Arch Nemesis: Mail Carrier" or "Parcel #5055"
+  description: string|null, // free-form description
+  media: string|null, // URL to associated media, preferably to decentralized, content-addressed storage
+  media_hash: string|null, // Base64-encoded sha256 hash of content referenced by the `media` field. Required if `media` is included.
+  issued_at: string|null, // When token was issued or minted, Unix epoch in milliseconds
+  expires_at: string|null, // When token expires, Unix epoch in milliseconds
+  starts_at: string|null, // When token starts being valid, Unix epoch in milliseconds
+  updated_at: string|null, // When token was last updated, Unix epoch in milliseconds
+  extra: string|null, // Anything extra the MT wants to store on-chain. Can be stringified JSON.
+  reference: string|null, // URL to an off-chain JSON file with more info.
+  reference_hash: string|null // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+type MTTokenMetadataAll = {
+  base: MTBaseTokenMetadata
+  token: MTTokenMetadata
+}
+```
+
+A new set of functions MUST be supported on the MT contract:
+
+```ts
+// Returns the top-level contract level metadtata
+function mt_metadata_contract(): MTContractMetadata {}
+function mt_metadata_token_all(token_ids: string[]): MTTokenMetadataAll[]
+function mt_metadata_token_by_token_id(token_ids: string[]): MTTokenMetadata[]
+function mt_metadata_base_by_token_id(token_ids: string[]): MTBaseMetadata[]
+function mt_metadata_base_by_metadata_id(base_metadata_ids: string[]): MTBaseTokenMetadata
+
+```
+
+A new attribute MUST be added to each `Token` struct:
+
+```diff
+ type Token = {
+   id: string,
++  token_metadata?: MTTokenMetadata,
++  base_metadata_id: string,
```

Comment:
What purpose does this serve?

Reply:
just realized that it doesn't show the line number i'm referring to - i'm asking about the `base_metadata_id` field

Reply:
So the premise is this. In the multi token standard there are many different tokens in a single contract. So metadata that would have been associated with a single contract now needs to be associated with a potential set  of token ids. 
```
MTBaseTokenMetadata {
 name: string, // required, ex. "Silver Swords" or "Metaverse 3"
  id: string, // required a unique identifier for the metadata
  symbol: string|null, // required, ex. "MOCHI"
  ....
}
```
^^ Represents the top level description 
```
MTTokenMetadata {
  title: string|null, // ex. "Arch Nemesis: Mail Carrier" or "Parcel #5055"
  description: string|null, // free-form description
  media: string|null, // URL to associated media, preferably to decentralized, content-addressed storage
...
}
```
^^ represents the instance specific data. 

In practice it looks like 
```
const baseMetadata = {
  name: "My Bird NFT Collection"
  symbol:  "MBNFT"
  id: "99"
}

const tokenMetadataOne = {
   title: "Raven #1"
   description: "Part of the My Bird NFT Collection - Raven"
}
const tokenMetadataTwo = {
  title: "Bluejay #2",
  description: " Part of the My Bird NFT Collection - Bluejay"
}
```
one token retrieval method then looks like 
```ts

function mt_token(token_ids: string[]): (Token | null)[]

mt_token(["1","2"]);

/* returns
 [{
   id: 1,
   token_metadata: {
    title: "Raven #1"
  } ,
  base_metadata_id: 99,
},
{
   id: 2,
   token_metadata: {
    title: "Bluejay #2"
  } ,
  base_metadata_id: 99,
}]
*/
```
Then retrieve base metadata for all the tokens fetched
```ts
function mt_metadata_base_by_metadata_id(base_metadata_ids:string[]): MTBaseTokenMetadata[]

mt_metadata_base_by_metadata_id("99"); 
/*returns
[{ 
  name: "My Bird NFT Collection"
  symbol:  "MBNFT"
  id: "99"
}]
*/
```